nsxt/resource_nsxt_logical_router_uplink_port_test.go

```diff
@@ -0,0 +1,188 @@
+					resource.TestCheckResourceAttr(testResourceName, "description", "Acceptance Test Updated"),
+					resource.TestCheckResourceAttr(testResourceName, "urpf_mode", "STRICT"),
+					resource.TestCheckResourceAttr(testResourceName, "tag.#", "2"),
+					resource.TestCheckResourceAttr(testResourceName, "subnets.#", "1"),
+					resource.TestCheckResourceAttr(testResourceName, "edge_cluster_member_index.#", "1"),
+					resource.TestCheckResourceAttrSet(testResourceName, "linked_logical_switch_port_id"),
+					resource.TestCheckResourceAttrSet(testResourceName, "logical_router_id"),
+				),
+			},
+		},
+	})
+}
+
+func testAccNsxtLogicalRouterUpLinkPortExists(displayName string, resourceName string) resource.TestCheckFunc {
+	return func(state *terraform.State) error {
+
+		nsxClient := testAccProvider.Meta().(nsxtClients).NsxtClient
+
+		rs, ok := state.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("NSX logical router uplink port resource %s not found in resources", resourceName)
+		}
+
+		resourceID := rs.Primary.ID
+		if resourceID == "" {
+			return fmt.Errorf("NSX logical router uplink port resource ID not set in resources ")
+		}
+
+		logicalPort, responseCode, err := nsxClient.LogicalRoutingAndServicesApi.ReadLogicalRouterUpLinkPort(nsxClient.Context, resourceID)
+		if err != nil {
+			return fmt.Errorf("Error while retrieving logical router uplink port ID %s. Error: %v", resourceID, err)
+		}
+
+		if responseCode.StatusCode != http.StatusOK {
+			return fmt.Errorf("Error while checking if logical router uplink port %s exists. HTTP return code was %d", resourceID, responseCode.StatusCode)
+		}
+
+		if displayName == logicalPort.DisplayName {
+			return nil
+		}
+		return fmt.Errorf("NSX logical port %s wasn't found", displayName)
+	}
+}
+
+func testAccNSXLogicalRouterUpLinkPortCheckDestroy(state *terraform.State, displayName string) error {
+	nsxClient := testAccProvider.Meta().(nsxtClients).NsxtClient
+	for _, rs := range state.RootModule().Resources {
+
+		if rs.Type != "nsxt_logical_router_uplink_port" {
+			continue
+		}
+
+		resourceID := rs.Primary.Attributes["id"]
+		logicalPort, responseCode, err := nsxClient.LogicalRoutingAndServicesApi.ReadLogicalRouterUpLinkPort(nsxClient.Context, resourceID)
+		if err != nil {
+			if responseCode.StatusCode != http.StatusOK {
+				return nil
+			}
+			return fmt.Errorf("Error while retrieving logical router uplink port ID %s. Error: %v", resourceID, err)
+		}
+
+		if displayName == logicalPort.DisplayName {
+			return fmt.Errorf("NSX logical router uplink port %s still exists", displayName)
+		}
+	}
+	return nil
+}
+
+func testAccNsxtLogicalRouterUpLinkPortPreconditionsTemplate(tier0RouterName string, transportZoneName string) string {
+	return fmt.Sprintf(`
+data "nsxt_logical_tier0_router" "tier0rtr" {
+  display_name = "%s"
+}
+
+data "nsxt_transport_zone" "tz1" {
+  display_name = "%s"
+}
+
+resource "nsxt_logical_switch" "ls1" {
```

Comment:
should this be nsxt_vlan_logical_switch, since it is defined in vlan transport zone according to the test?